coalib/core/Graphs.py

```diff
@@ -0,0 +1,62 @@
+from coalib.core.CircularDependencyError import CircularDependencyError
+
+
+def traverse_graph(start_nodes, get_successive_nodes,
+                   run_on_edge=lambda prev, nxt: None):
+    """
+    Traverses all edges of a directed acyclic graph once. Detects cyclic graphs
```

Comment:
Actually I misinterpreted that statement. and yes rest of the statement needs the change.